### PR TITLE
Add support for integration events

### PR DESCRIPTION
**Pull request by Suhaibinator**
Discordgo does not currently support Integration creates, updates, and deletes. This PR adds support for that.

## See also
- Gateway intents (1 << 4) [Gateway intents](https://discord.com/developers/docs/topics/gateway)
- [Integration events](https://discord.com/developers/docs/topics/gateway-events#integration-create)

### CODE REVIEW

**Comment by Suhaibinator** [comment]
@FedorLap2006

File: events.go [hunk 1]
```diff
@@ -241,6 +241,21 @@ type GuildScheduledEventUserRemove struct {
 	GuildID               string `json:"guild_id"`
 }
 
+// IntegrationCreate is the data for a IntegrationCreate event.
+type IntegrationCreate struct {
+	*Integration
```
Comment (FedorLap2006) [inline events.go:246]: Payload also contains `guild_id`

File: events.go [hunk 2]
```diff
@@ -241,6 +241,21 @@ type GuildScheduledEventUserRemove struct {
...
+}
+
+// IntegrationUpdate is the data for a IntegrationUpdate event.
+type IntegrationUpdate struct {
+	*Integration
```
Comment (FedorLap2006) [inline events.go:251]: `guild_id` is here too

File: events.go [hunk 3]
```diff
@@ -241,6 +241,21 @@ type GuildScheduledEventUserRemove struct {
...
+}
+
+// IntegrationDelete is the data for a IntegrationDelete event.
+type IntegrationDelete struct {
+	*Integration
```
Comment (FedorLap2006) [inline events.go:256]: The payload isn't an `Integration` object. Rather it's
```suggestion
    ID                   string `json:"id"`
    GuildID           string `json:"guild_id"`
    ApplicationID string `json:"application_id,omitempty"`
```

Comment (FedorLap2006) [inline events.go:256]: The payload isn't an `Integration` object. Rather it's
```suggestion
    ID string `json:"id"`
    GuildID string `json:"guild_id"`
    ApplicationID string `json:"application_id,omitempty"`
```
P.S. This is unformatted.

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]

File: events.go (hunk 1 above)
Comment (Suhaibinator) [reply events.go:246]: Done

File: events.go (hunk 2 above)
Comment (Suhaibinator) [reply events.go:252]: Done

**Comment by Suhaibinator** [comment]
@FedorLap2006 requested changes have been made

**Review by FedorLap2006 — APPROVED** [review]

**Comment by Suhaibinator** [comment]
@FedorLap2006 do you know why it is saying that the format check failed?

Also it seems I am not able to merge the PR. Does the merge have to be performed by a maintainer?

**Pull request merged**